build/buildall

```diff
@@ -246,3 +250,40 @@ time (
     -Dbuildver="$BASE_VER" \
     -DskipTests -Dskip -Dmaven.javadoc.skip
 )
+
+###############################################################################################
+# the following is to check the revisions in shims are identical
+###############################################################################################
+shims_for_check="${SPARK_SHIM_VERSIONS[@]}"
+if [[ "$with_db_build" == "true" ]]; then
+  shims_for_check="${shims_for_check} 301db 312db"
+fi
+
+echo "build all finished"
+
+# PWD should be spark-rapids root path
+parallel_dir=${PWD}/dist/target/parallel-world
+pre_revision=""
+pre_shim_version_path=""
+for shim in ${shims_for_check}; do
+  shim_version_path="${parallel_dir}/spark${shim}/rapids4spark-version-info.properties"
+  if [[ -f "$shim_version_path" ]] ; then
+    curr_revision=$(cat ${shim_version_path} | grep "revision=" | cut -d'=' -f2)
+    if [ -n "$pre_revision" ] && [[ "$curr_revision" != "$pre_revision" ]] ; then
+      echo
+      echo "=================================================================="
+      echo "Check Failed!!!: revisions are not equals"
+      echo "=================================================================="
+      echo "  in path ${pre_shim_version_path} ${pre_revision}"
+      echo "  in path ${shim_version_path} ${curr_revision}"
+      echo "Although the is build finished, but you are taking the risk of inconsistent revisions of shims!!!"
```

Comment:
I think I would simplify this and just say something like:
Please check the revisions of each shim to see which one is inconsistent.  Note,  if building with Databricks those jars are built separately.